frontend/storage/reducers/gameVisibilityReducer.ts

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-unused-vars
+import {AppStore, SET_GAME_VISIBILITY, AppActions} from '../common';
+
+export function gameVisibilityReducer(state: AppStore, action: AppActions): AppStore {
+	switch (action.type) {
+		case SET_GAME_VISIBILITY: {
+			if (state.gamesVisibility.get(action.payload.gameId)) {
+				const newGamesVisibility = new Map(state.gamesVisibility);
+				newGamesVisibility.set(action.payload.gameId, action.payload.gameVisibility);
+				return {
+					...state,
+					gamesVisibility: new Map(newGamesVisibility),
```

Comment:
You don't need to create another map here, because `newGamesVisibility` is already a clone of the one stored in the state!